repotracker/repotracker.go

```diff
@@ -73,7 +73,12 @@ func (p *ProjectInfo) notPopulated() bool {
 
 // PopulateVersion updates the version's ParserProject if we have or can create it
 func (p *ProjectInfo) populateVersion(v *model.Version) error {
-	if p.IntermediateProject != nil {
+	if evergreen.UseParserProject && p.IntermediateProject != nil {
+		config, err := yaml.Marshal(p.IntermediateProject)
+		if err != nil {
+			return errors.Wrap(err, "error marshalling intermediate project")
+		}
+		v.Config = string(config)
 		v.ParserProject = p.IntermediateProject
 		return nil
 	}
```

Comment:
Shouldn't the line below this one use the config yaml from the version, not a config string generated from marshaling the project? Or was this project created from the config file originally?

Reply:
This function is used when creating a new version from the repotracker, and so the version config yaml is not yet populated and is being created from this project.

Reply:
However, before this change we read from the completed project (I believe because in some cases we have that and not the IntermediateProject). I'll look at this again.

Reply:
Ah, I see. LGTM.